src/main/scala/com/databricks/spark/redshift/Utils.scala

```diff
@@ -60,6 +60,42 @@ private[redshift] object Utils {
   }
 
   /**
+   * Factory method to create new S3URI in order to handle various library incompatabilities with
+   * older AWS Java Libraries
+   */
+  def createS3URI(url: String): AmazonS3URI = {
+    try {
+      // try to instantiate AmazonS3URI with url
+      new AmazonS3URI(url)
+    } catch {
+      case e: java.lang.IllegalArgumentException => {
```

Comment:
To avoid having to re-throw exceptions which don't have the correct message, you can combine the `if` with the `case` statement, e.g. `case e: IllegalArgumentException if e.getMessage.startsWith(...) =>`.